Sources/MUXSDKStats/MUXSDKPlayerBinding.m

```diff
@@ -539,6 +593,67 @@ - (CGSize)getSourceDimensions {
     return CGSizeMake(0, 0);
 }
 
+- (long)findMostRecentAdvertisedBitrateForPlayerItem:(nonnull AVPlayerItem *)item {
+    AVPlayerItemAccessLog *accessLog = [item accessLog];
+    
+    if (accessLog && accessLog.events) {
+        NSArray<AVPlayerItemAccessLogEvent *> *events = [accessLog.events copy];
+        
+        AVPlayerItemAccessLogEvent *lastEvent = [events lastObject];
+        if (lastEvent) {
+            long long transferredBytes = lastEvent.numberOfBytesTransferred;
+            NSTimeInterval downloadedDuration = lastEvent.segmentsDownloadedDuration;
+            long long transferredBits = transferredBytes * 8;
+            double calculatedBitrate = transferredBits / downloadedDuration;
+            return (long)calculatedBitrate; // any rounding is fine
+        } else {
+            NSLog(@"findMostRecentAdvertisedBitrateForPlayerItem: no events");
+        }
+    }
+    
+    return -1;
+}
+
+- (nullable NSString *)getURLStringIfAvailableOnAVPlayerItem:(nonnull AVPlayerItem *)item {
+    AVAsset *playerItemAsset = item.asset;
+    if (playerItemAsset && [playerItemAsset isKindOfClass:AVURLAsset.class]) {
+        AVURLAsset *urlAsset = (AVURLAsset *)playerItemAsset;
+        NSString * urlString = [[urlAsset URL] absoluteString];
+        return urlString;
+    }
+    
+    return nil;
+}
+
+- (nullable AVPlayerItemTrack *)findVideoTrackOnPlayerItem:(nonnull AVPlayerItem *)item {
+    if (!item.tracks) {
+        NSLog(@"findVideoTrackOnPlayerItem: no tracks loaded yet");
+        return nil;
+    }
+    
+    // TODO: what even throws here anyway? Is this catch-block really needed?
+//    @try {
+        for (int t = 0; t < item.tracks.count; t++) {
+            AVPlayerItemTrack *playerItemTrack = [
+                [item tracks] objectAtIndex:t
+            ];
+            if (playerItemTrack) {
+                AVAssetTrack *assetTrack = playerItemTrack.assetTrack;
+                if (assetTrack) {
+                    AVMediaType mediaType = assetTrack.mediaType;
+                    if (mediaType && [mediaType isEqualToString:AVMediaTypeVideo]) {
+                        return playerItemTrack;
+                    }
+                }
+            }
+        }
+//    } @catch (NSException *exception) {
+//        NSLog(@"%@", exception.reason);
+//    }
```

Comment:
Added a note to the other try/catch, but here it's still possible (if extremely unlikely) that the `tracks` property could change out from under you. In any case, `for in` loops are easier to write:
```suggestion
    for (AVPlayerItemTrack *playerItemTrack in item.tracks) {
        if ([playerItemTrack.assetTrack.mediaType isEqual:AVMediaTypeVideo]) {
            return playerItemTrack;
        }
    }
```
I also suggested here to use `isEqual:` instead of `isEqualToString:`. The only "benefit" of `isEqualToString:` is that it throws an exception if the argument or receiver is not a string. So it's almost always a good idea to just use `isEqual:`.